beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/service/BeekeeperHistoryService.java

```diff
@@ -0,0 +1,47 @@
+package com.expediagroup.beekeeper.core.service;
+
+import java.time.LocalDateTime;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingEntity;
+import com.expediagroup.beekeeper.core.model.history.BeekeeperHistory;
+import com.expediagroup.beekeeper.core.repository.BeekeeperHistoryRepository;
+
+public class BeekeeperHistoryService {
+
+  private static final Logger log = LoggerFactory.getLogger(BeekeeperHistoryService.class);
+
+  private final BeekeeperHistoryRepository beekeeperHistoryRepository;
+
+  public BeekeeperHistoryService(BeekeeperHistoryRepository beekeeperHistoryRepository) {
+    this.beekeeperHistoryRepository = beekeeperHistoryRepository;
+  }
+
+  public void saveHistory(HousekeepingEntity housekeepingEntity, String status) {
+    String eventDetails = getEventDetails(housekeepingEntity);
+
+    BeekeeperHistory event = BeekeeperHistory.builder()
+        .id(housekeepingEntity.getId())
+        .eventTimestamp(LocalDateTime.now())
+        .databaseName(housekeepingEntity.getDatabaseName())
+        .tableName(housekeepingEntity.getTableName())
+        .lifecycleType(housekeepingEntity.getLifecycleType())
+        .housekeepingStatus(status)
+        .eventDetails(eventDetails)
+        .build();
+
+    log.info("Saving activity in Beekeeper History table; {}", event);
+    beekeeperHistoryRepository.save(event);
+  }
+
+  private String getEventDetails(HousekeepingEntity entity) {
+    return entity.toString();
```

Comment:
I'm not sure if this method generate to much value, since it just call toString method of class

Reply:
I disagree, the point of this is to know the details of exactly what was saved/failed as a history, so all of the fields in the HousekeepingEntity are relevant here. 
By doing it like this, we can easily get all the details from Housekeeping metadata, path, and any future types without having to change it to pull out specific parts of each new type. 
In addition, if we only pull out some parts of the entity, we may find that later we want to have more detail. By having the toString, we get a simple output with all the details. 
 
Example of logs:
```
Saving activity in Beekeeper History table; BeekeeperHistory(id=1, eventTimestamp=2024-11-25T17:42:20.605450, databaseName=some_database, tableName=some_table, lifecycleType=UNREFERENCED, housekeepingStatus=DELETED, eventDetails=HousekeepingPath(path=s3://test-path-bucket/some_database/some_table/ /id1/partition1, databaseName=some_database, tableName=some_table, housekeepingStatus=DELETED, creationTimestamp=2024-11-25T17:28:33, modifiedTimestamp=2024-11-25T17:28:33, cleanupTimestamp=2024-11-24T17:28:34, cleanupDelay=PT1S, cleanupAttempts=1, clientId=client_id, lifecycleType=UNREFERENCED))
```

Reply:
yes, I got you point. I agree to call toString(), I'm not sure about to create a method just to call toString method getEventDetails